### DISCUSSION TITLE
Can't build on msys2 [ucrt]

### DISCUSSION THREAD

**Discussion by LinArcX** [leo-arch/clifm#283] [Q&A] [answered]
Here are all the packages that i have already on my machine:
```
base 2022.06-1
filesystem 2023.02.07-1
gettext-devel 0.22.4-1
git 2.44.0-1
glib2-devel 2.78.4-1
libreadline-devel 8.2.010-1
make 4.4.1-2
mingw-w64-ucrt-x86_64-gettext-tools 0.22.5-2
mingw-w64-ucrt-x86_64-neovim 0.9.5-3
mingw-w64-ucrt-x86_64-pkg-config 0.29.2-6
mingw-w64-ucrt-x86_64-readline 8.2.010-1
mingw-w64-ucrt-x86_64-repgrep 0.15.0-1
mingw-w64-ucrt-x86_64-rust 1.77.2-1
msys2-runtime 3.4.10-8
msys2-runtime-devel 3.4.10-8
```

When i run: ` make -f misc/cygwin/Makefile` i recive tons of errros like this: 

`fatal error: glob.h: No such file or directory`

**Comment by leo-arch** [comment]
`/usr/include/glob.h` is provided by the `msys2-runtime-devel` package, which you have already installed. Please make sure the file is actually there.

Try building as follows:

```
cd clifm
gcc -o clifm.exe src/*.c -lreadline -lacl -lmagic -lintl
```

**Reply by LinArcX** [reply]
no, still have the issue.

**Reply by leo-arch** [reply]
Can you confirm the existence of `/usr/include/glob.h`?

**Reply by LinArcX** [reply]
Yes, it's there.

**Reply by leo-arch** [reply]
Please copy here the exact error messages. Maybe there's something else.

**Reply by LinArcX** [reply]
https://gist.github.com/LinArcX/5200f99dbd4041917111eab092906fc7

**Reply by leo-arch** [reply]
You're definitely missing the file. I assume the appropriate package isn't installed.

@muellerto, some hint on this?

**Reply by LinArcX** [reply, marked as answer]
I got it!

I was runnnig `make` in MSYS2 UCRT64 environment. and this one by default looks for libs in: `/ucrt64/include/` and the interesting thing is that, if you install `msys2-runtime-devel` it will put their files in `/usr/include`. So i changed my msys2 terminal to MSYS2 MSYS and tried again. it built successfully.

**Reply by leo-arch** [reply]
Glad it's solved.

**Comment by muellerto** [comment]
Need about 14 hours for this ... My Windows time for today is over (writing this on Linux).

**Comment by muellerto** [comment]
Ah, good. 

@LinArcX I use the `mintty` terminal on Windows. This is included in MSYS or at least you can install it with `pacman` there. This is more handy and very solid. Note: some configuration (font?) must be done in a rc file.
